semanticstack-rql-parser/src/main/java/com/boschsemanticstack/rql/parser/v1/RqlParser.java

```
@@ -45,6 +56,23 @@ public static RqlQueryModel from( final String rqlQuery ) {
       return new RqlParserApi().parseFullQuery( rqlQuery );
    }
 
+   public static RqlQueryModel preProcessFilter( final RqlQueryModel model ) {
+      return preProcessFilter( model, DEFAULT_FILTER_PREPROCESSORS );
+   }
+
+   public static RqlQueryModel preProcessFilter(
+         final RqlQueryModel model,
+         final List<RqlFilterPreProcessor> filterPreProcessors ) {
+      if ( model == null ) {
+         return null;
```

Do you think maybe return Optional make sense or maybe throw a exception?

---

Makes sense, I will throw an exception!